lib/libsmb2.c

```diff
@@ -465,7 +465,7 @@ smb2_opendir_async(struct smb2_context *smb2, const char *path,
         if (pdu == NULL) {
                 free_smb2dir(smb2, dir);
                 smb2_set_error(smb2, "Failed to create opendir command.");
-                return -1;
+                return -EINVAL;
```

Comment:
smb2_cmd_create_async can fail because of mem allocation or can fail because of failure during encoding, so I feel EINVAL is more appropriate Let me know your thoughts.